client/src/components/molecules/DonationItemForm.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { FunctionComponent, useState } from "react";
+
+import DonationForm, {
+    ItemAgeToDescriptionMap,
+    ItemCondition,
+    ItemConditionToDescriptionMap
+} from "../../data/types/donationForm";
+import { Button } from "../atoms/Button";
+import FormItem from "../molecules/FormItem";
+import RequestGroup from "../../data/types/requestGroup";
+import ScrollableDropdown from "../atoms/ScrollableDropdown";
+import SearchableDropdown from "../atoms/SearchableDropdown";
+import TextArea from "../atoms/TextArea";
+import { TextField } from "../atoms/TextField";
+
+interface Props {
+    onDelete: () => void;
+    onSave: (donationForm: DonationForm) => void;
+    requestGroups: Array<RequestGroup>;
+    showFormUnsavedError: boolean;
+}
+
+const DonationItemForm: FunctionComponent<Props> = (props: Props) => {
+    const [name, setName] = useState("");
+    const [nameInput, setNameInput] = useState("");
+    const [condition, setCondition] = useState<ItemCondition | null>(null);
+    const [age, setAge] = useState(1);
+    const [quantity, setQuantity] = useState<number>(1);
+    const [description, setDescription] = useState<string>("");
+    const [nameError, setNameError] = useState("");
+    const [formError, setFormError] = useState("");
+    const [isConditionErrorneous, setIsConditionErrorneous] = useState(false);
+    const [isQuantityErrorneous, setIsQuantityErrorneous] = useState(false);
```

Comment:
nit: should be "Erroneous" instead of "Errorneous"

Reply:
LOL good catch